Lib/test/test_set.py

```diff
@@ -667,10 +667,12 @@ def __new__(cls, arg, newarg=None):
                 self = super().__new__(cls, arg)
                 self.newarg = newarg
                 return self
-        u = subclass_with_new([1, 2], newarg=3)
+        u = subclass_with_new([1, 2])
         self.assertIs(type(u), subclass_with_new)
         self.assertEqual(set(u), {1, 2})
-        self.assertEqual(u.newarg, 3)
+        self.assertIsNone(u.newarg)
+        with self.assertRaises(TypeError):
```

Comment:
Consider adding a comment linking to [the message that inspired this decision](https://bugs.python.org/issue43413#msg402000).

Reply:
What comment exactly? "Raymond said to not support this"?

Reply:
```suggestion
        # disallow kwargs in __new__ only (https://bugs.python.org/issue43413#msg402000)
        with self.assertRaises(TypeError):
```